src/api/app/models/patchinfo.rb

```diff
@@ -257,4 +256,106 @@ def issues_by_tracker
     end
     issues_by_tracker
   end
+
+  def load_from_xml(patchinfo_xml)
+    self.binaries = []
+    patchinfo_xml.elements('binary').each do |binaries|
+      self.binaries << binaries
+    end
+    self.packager = patchinfo_xml.value('packager')
+    self.version = patchinfo_xml['version']
+
+    self.issues = []
+    patchinfo_xml.elements('issue') do |issue_element|
+      if issue_element['_content'].blank?
+        # old uploaded patchinfos could have broken tracker-names like "bnc "
+        # instead of "bnc". Catch these.
+        begin
+          summary = IssueTracker::IssueSummary.new(issue_element['tracker'], issue_element['id'])
+          issue_element['_content'] = summary.issue_summary if summary.belongs_bug_to_tracker?
+        rescue Backend::NotFoundError
+          issue_element['_content'] = 'PLEASE CHECK THE FORMAT OF THE ISSUE'
+        end
+      end
+
+      issue_tracker = IssueTracker.find_by_name(issue_element['tracker']).
+                      try(:show_url_for, issue_element['id']).to_s
+
+      issues << [
+        issue_element['tracker'],
+        issue_element['id'],
+        issue_tracker,
+        issue_element['_content']
+      ]
+    end
+    self.category = patchinfo_xml.value('category')
+    self.rating = patchinfo_xml.value('rating')
+    self.summary = patchinfo_xml.value('summary')
+    self.name = patchinfo_xml.value('name')
+
+    self.description = patchinfo_xml.value('description')
+    self.message = patchinfo_xml.value('message')
+    self.relogin_needed = !patchinfo_xml.value('relogin_needed').nil?
+    self.reboot_needed = !patchinfo_xml.value('reboot_needed').nil?
+    self.zypp_restart_needed = !patchinfo_xml.value('zypp_restart_needed').nil?
+    if patchinfo_xml.value('stopped')
+      self.block = true
+      self.block_reason = patchinfo_xml.value('stopped')
+    end
+
+    self
+  end
+
+  def to_xml(project, package)
```

Comment:
[nitpick] if the to_xml is in the model, you can (and possibly should) use CanRenderModel and builder file